kong/tools/wrpc/proto.lua

```diff
@@ -0,0 +1,164 @@
+local pb = require "pb"
+local grpc = require "kong.tools.grpc"
+
+local grpc_new = grpc.new
+local pb_encode = pb.encode
+local setmetatable = setmetatable
+local string_format = string.format
+
+local _M = {}
+local _MT = { __index = _M, }
+
+local wrpc_proto_name = "wrpc.wrpc"
+
+local default_proto_path = { "kong/include/", "/usr/include/", }
+
+local function parse_annotation(annotation)
+  local parsed = {}
+  for kv_pair in annotation:gmatch("[^;]+=[^;]+") do
+    local key, value = kv_pair:match("^%s*(%S-)=(%S+)%s*$")
+    parsed[key] = value
```

Comment:
this would give a hard error if there's no match

Reply:
Seems like an incorrect manually solved merge...

Reply:
You are right. if `kv_pair` happens to be `" = "` the `%S+` fails to match.
I'm not sure why we match it this way(`(%S-)` for key and `(%S+)` for value). Is it ok for the key to be `""`?

Reply:
`(%S+)` would swallow the `=` and then do a double backtrack loop from the end (one for the `(%S+)` and another for the `%s*` after it) to find it again, resulting in a O(n^2) behaviour instead of 2*O(n).  another way to do it would be `^%s*([^%s=]+)=(%S+)`; better but that creates a match table on each invocation.